blocks_vertical/procedures.js

```diff
@@ -60,6 +60,15 @@ Blockly.Blocks['procedures_callnoreturn'] = {
     this.procCode_ = '';
   },
   /**
+   * Returns the name of the procedure this block calls, or the empty string if
+   * it has not yet been set.
+   * @return {string} Procedure name.
+   * @this Blockly.Block
+   */
+  getProcCode: function() {
+    return this.procCode_;
```

Comment:
Question: I see procCode is defined in the mutation XML as well as stored in this variable. Which is the "source of truth"? What is the lifecycle of the mutation XML? Is that only used when loading/serializing the block?

Reply:
https://developers.google.com/blockly/guides/create-custom-blocks/web/mutators

The XML is used when loading and serializing the block.  We don't keep a handle on it while the block is alive.  In this case the XML is how scratch-vm is telling scratch-blocks what to load.